src/AI_ExtensibleSaveFormat/AI.ExtendedSave.Hooks.cs

```diff
@@ -14,11 +22,248 @@ internal static partial class Hooks
             [HarmonyPatch(typeof(GameCoordinateFileInfoAssist), nameof(GameCoordinateFileInfoAssist.CreateCoordinateFileInfoList))]
             private static void CreateListPostfix() => LoadEventsEnabled = true;
+
+
+            public static void InstallMainGameHooks(Harmony harmony)
+            {
+                // Minimizing the interference between games to limiting the harmony hook initialization
+                // MainGame Save/Load - PostFix
+                harmony.Patch(
+                    typeof(SaveData).GetMethod("Load", new[] {typeof(BinaryReader)}),
+                    null, new HarmonyMethod(typeof(Hooks).GetMethod(nameof(OnLoadMainGame)))
+                );
+                harmony.Patch(
+                    typeof(SaveData).GetMethod("SaveFile", new[] {typeof(BinaryWriter)}),
+                    null, new HarmonyMethod(typeof(Hooks).GetMethod(nameof(OnSaveMainGame)))
+                );
+
+                // HousingData Save/Load - Transpiler
+                // Based on 2021 Version
+                harmony.Patch(
+                    typeof(CraftInfo).GetMethod("Load", new[] {typeof(string)}),
+                    null, null, new HarmonyMethod(typeof(Hooks).GetMethod(nameof(OnCraftLoad)))
+                );
+                harmony.Patch(
+                    typeof(CraftInfo).GetMethod("Save", new[] {typeof(string), typeof(byte[])}),
+                    null, null, new HarmonyMethod(typeof(Hooks).GetMethod(nameof(OnCraftSave)))
+                );
+            }
+
+            /// <summary>
+            ///
+            /// </summary>
+            /// <param name="saveData"></param>
+            /// <param name="br"></param>
+            public static void GameDataLoadHook(AIProject.SaveData.SaveData saveData, BinaryReader br)
+            {
+                try
+                {
+                    string marker = br.ReadString();
+                    int version = br.ReadInt32();
+                    int length = br.ReadInt32();
+                    if (marker == Marker && version == DataVersion && length > 0)
+                    {
+                        ReadWorldData(saveData, br);
+                        Logger.LogMessage("loaded game data extended");
+                    }
+                    else Logger.LogMessage("empty game data extended");
+                }
+                catch (EndOfStreamException)
+                {
+                } //Incomplete/non-existant data
+                catch (System.SystemException)
+                {
+                } //Invalid/unexpected deserialized data
+            }
+
+            /// <summary>
+            ///
+            /// </summary>
+            /// <param name="saveData"></param>
+            /// <param name="br"></param>
+            public static void ReadWorldData(SaveData saveData, BinaryReader br)
+            {
+                var marker = br.ReadString();
+                if (marker != "WD") return;
```

Comment:
Use a constant for the marker

Reply:
Well "WD" was totally random token in my mind when i wrote this one. I'll try to make it as constant when there is no other better way to separate world data

Reply:
https://github.com/IllusionMods/BepisPlugins/pull/122/commits/a57e34b7c5f0c984a134ea1e0023d8c6bfee7d86